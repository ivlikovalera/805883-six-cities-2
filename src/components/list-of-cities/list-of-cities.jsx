import React from "react";
import City from './../city/city.jsx';
import {PropTypes as pt} from 'prop-types';
import {connect} from 'react-redux';
import {getUniqueCities} from './../../reducer/data/selector.js';

export const ListOfCities = (props) => {
  const {uniqueCities} = props;
  return <section className="locations container">
    <ul className="locations__list tabs__list">
      {uniqueCities.map((it, i) => <li className="locations__item" key={i}>
        <City
          key={i}
          name={it.name}
        />
      </li>)}
    </ul>
  </section>;
};

ListOfCities.propTypes = {
  uniqueCities: pt.arrayOf(pt.shape({
    name: pt.string.isRequired,
  })),
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  uniqueCities: getUniqueCities(state),
});

export default connect(mapStateToProps)(ListOfCities);
