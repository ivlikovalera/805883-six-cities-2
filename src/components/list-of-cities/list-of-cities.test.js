import React from 'react';
import renderer from 'react-test-renderer';
import {ListOfCities} from './list-of-cities.jsx';

it(`List of cities correctly renders after relaunch`, () => {

  const listOfCitiesComponent = renderer
  .create(<ListOfCities
    uniqueCities={[]}
  />)
  .toJSON();

  expect(listOfCitiesComponent).toMatchSnapshot();
});
