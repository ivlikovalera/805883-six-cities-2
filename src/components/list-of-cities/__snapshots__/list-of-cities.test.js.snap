// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List of cities correctly renders after relaunch 1`] = `
<section
  className="locations container"
>
  <ul
    className="locations__list tabs__list"
  />
</section>
`;
